app/views/preferences/edit.html.erb

```diff
@@ -0,0 +1,12 @@
+<% content_for :heading do %>
+  <h1><%= t ".title" %></h1>
+<% end %>
+
+<%= bootstrap_form_for current_user, :url => { :action => :update } do |f| %>
+  <%= f.select :preferred_editor, [[t("editor.default", :name => t("editor.#{Settings.default_editor}.name")), "default"]] + Editors::AVAILABLE_EDITORS.collect { |e| [t("editor.#{e}.description"), e] } %>
+
+  <%= f.text_field :languages %>
+
+  <%= f.primary t(".save") %>
+  <%= link_to t(".cancel"), preferences_path, :class => "btn btn-link" %>
```

Comment:
Would `btn-outline-secondary` be a better a choice here? Having a link alongside a button looks a little odd to me?

Reply:
I'm happy with the cancel link being just a link, since it reinforces the idea that it's not going to change any state on the server. 

We also use this pattern elsewhere already (traces, user blocks and messages) so I'd prefer to keep it consistent here, and if we do decide to change it, do so everywhere in a separate PR.

Reply:
Yes I did notice that pattern was being used in other places - maybe I'll just do a PR to fix them all then.